DiamondLightSource/SynchWeb
# LIMS-1341: Allow entry of SMILES codes

**JIRA ticket**: [LIMS-1341](https://jira.diamond.ac.uk/browse/LIMS-1341)

**Summary**:

Add an editable box for SMILES code. Regex is fun as lots of characters are allowed, see https://en.wikipedia.org/wiki/Simplified_molecular-input_line-entry_system

**Changes**:
- Add entry box to plate sample page
- Add code to allow editing in the database, and get current value

**To test**:
- Go to proposal nt37104, go to Containers, click on a plate
- Check the SMILES code box appears under sample
- Edit the SMILES code, using weird symbols eg `.:%=#$@+-[]()/\`, check it remains after a refresh
- Change sample, check the value changes as expected.
